QAchat/py/QAchat.ipynb

```diff
@@ -0,0 +1,361 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# BrainTrust QA Chat Tutorial\n",
+    "\n",
+    "<a target=\"_blank\" href=\"https://colab.research.google.com/github/braintrustdata/braintrust-examples/blob/main/QAchat/py/QAchat.ipynb\">\n",
+    "  <img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/>\n",
+    "</a>\n",
+    "\n",
+    "Welcome to [BrainTrust](https://www.braintrustdata.com/)! This is a quick tutorial on how to build and evaluate an AI question and answer chat assistant. The assistant answers questions based on the user's information that is already saved in a vector DB (Chroma).\n",
+    "\n",
+    "Before starting, make sure that you have a BrainTrust account. If you do not, please [sign up](https://www.braintrustdata.com) or [get in touch](mailto:info@braintrustdata.com). After this tutorial, learn more by visiting [the docs](http://www.braintrustdata.com/docs)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# NOTE: Replace YOUR_OPENAI_KEY with your OpenAI API Key and YOUR_BRAINTRUST_API_KEY with your BrainTrust API key. Do not put it in quotes.\n",
+    "%env OPENAI_API_KEY=sk-XXXXX\n",
+    "%env BRAINTRUST_API_KEY=sk-XXXXX\n",
+    "%env TOKENIZERS_PARALLELISM=true # This is needed to avoid a warning message from Chroma"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install braintrust openai autoevals chromadb"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import braintrust\n",
+    "import chromadb\n",
+    "import openai\n",
+    "import asyncio\n",
+    "from autoevals.string import *\n",
+    "from autoevals.llm import *\n",
+    "\n",
+    "PROJECT_NAME = \"QAchatbot\""
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Load in datasets\n",
+    "\n",
+    "First, we'll load two datasets:\n",
+    "1. An evaluation dataset to test our pipeline. This includes input, output pairs like:\n",
+    "```\n",
+    "    {\"input\": \"What is my full name?\", \"output\": \"John Smith. -BT\"}\n",
+    "```\n",
+    "2. A user context dataset to give to the AI assistant as context. This will be stored in a vector db and contains rows like:\n",
+    "```\n",
+    "    {\"category\": \" address\", \"detail\": \"123 Main Street, Anytown, USA\"}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
create a separate block before the braintrust code that defines `eval_set` and `user_context_dataset`

Then, the block that calls into braintrust will be very legible/clear.